### PR TITLE
Add Typescript definitions

### PR DESCRIPTION
**Pull request by freakinruben**
This PR adds a TS definitions file

### CODE REVIEW

**Comment by rlepinski** [comment]
@freakinruben This looks great. Do you know of any tools that we can use to keep this file in sync with our other source? How are you validating your ts is correct?

**Comment by freakinruben** [comment]
I've searched for automatic tools but didn't find any (only the other way around).. Current file is validated with my current code-base (although I'm not using all features so it might not be fully correct).

The reason I hope this can be merged is also since it can't be done automatically (yet). Keeping them in sync while living in a different repo is a bigger hassle.

File: js/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,433 @@
...
+    /** The message sent date in milliseconds. **/
+    sentDate: number;
+    /** ptional - The icon url for the message. **/
+    listIconUrl: string;
+    /** The unread / read status of the message. **/
+    isRead: boolean;
+    /** The deleted status of the message. **/
+    isDeleted: boolean;
+    /** String to String map of any message extras. **/
+    extras: Map<string, any>;
```
Comment (rlepinski) [inline js/index.d.ts]: Can be updated to  <string, string>

**Comment by rlepinski** [comment]
Looks pretty good. One minor comment. Could you fill out https://docs.urbanairship.com/contribution-agreement/ ? Thanks!

**Comment by freakinruben** [comment]
@rlepinski After filling out the form I get:
> Your account has been disabled. Please contact Pardot Support (http://help.pardot.com/customer/en/portal/articles/2133457-how-do-i-contact-pardot-support-) to have your account re-enabled.

**Comment by rlepinski** [comment]
@freakinruben Trying to resolve the contribution agreement issue. Ill keep you posted.

**Comment by rlepinski** [comment]
@freakinruben New [contribution agreement](https://docs.google.com/forms/d/e/1FAIpQLScErfiz-fXSPpVZ9r8Di2Tr2xDFxt5MgzUel0__9vqUgvko7Q/viewform)

**Comment by freakinruben** [comment]
@rlepinski signed ;)

**Comment by rlepinski** [comment]
Thanks!

**Pull request merged**